mobile/test/kotlin/integration/EngineApiTest.kt

```diff
@@ -21,6 +24,29 @@ class EngineApiTest {
       EngineBuilder()
         .addLogLevel(LogLevel.INFO)
         .addStatsFlushSeconds(1)
+        .setNodeId("node-id")
+        .setNodeLocality("region", "zone", "subzone")
+        .setNodeMetadata(
+          Struct.newBuilder()
+            .putFields("string_value", Value.newBuilder().setStringValue("string").build())
+            .putFields("number_value", Value.newBuilder().setNumberValue(123.0).build())
+            .putFields("bool_value", Value.newBuilder().setBoolValue(true).build())
+            .putFields("null_value", Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build())
+            .putFields(
+              "struct_value",
+              Value.newBuilder()
+                .setStructValue(
+                  Struct.newBuilder()
+                    .putFields(
+                      "nested_value",
+                      Value.newBuilder().setStringValue("nested_string").build()
+                    )
+                    .build()
```

Comment:
I'm surprised to see multiple calls to `build()`. Are they required? Also, should we assert something to verify that the metadata is set correctly?

Reply:
Unfortunately, the Java proto API is verbose. It requires multiple calls of `build()` :(

I'm not aware of a way to assert the bootstrap config unless we expose that in the `EngineBuilder`. So for this test, it's more to ensure the JNI code doesn't crash. As for testing whether or not `node.metadata` is set, I tested it by printing the `Bootstrap` configuration via `bootstrap->PrintDebugString()`.

Reply:
I added a unit test for this. Hopefully that's good enough.